### ISSUE TITLE
Hugo quietly builds empty index.html files

### ISSUE THREAD

**Issue by SvenDowideit** [spf13/hugo#1659]
I'm finding an odd functionality.

if we have a content subdir with no `index.md` in it, `hugo server` shows 404, but hugo actually seems to generate an `index.html` using the `list.html` layout template.

as an extra bonus problem, if we remove all the `list.html`s, Hugo warns us __BUT ONLY IN VERBOSE MODE_, but when generating the files for upload to s3, it generates an empty `index.html` file.

I think there are a couple of issues really
1 `hugo server` is not sufficiently identical to the files that are generated
2 `hugo` build is very quiet about layout/template problems by default
3 `hugo` build happily writes out zero length files without error.

I'm guessing that 1&3 would be sufficient for what I need

```
docker run --rm -it  -e AWS_S3_BUCKET -e NOCACHE --name docker-docs-tools -p 8000:8000 -e DOCKERHOST "docs-base:master" hugo server --port=8000 --baseUrl=localhost --bind=0.0.0.0 --config=./config.toml  -v --verboseLog
Logging to /tmp/hugo022571655
INFO: 2015/12/02 Using config file: ./config.toml
INFO: 2015/12/02 syncing from /docs/themes/hugo-foundation/static to /docs/public/
INFO: 2015/12/02 syncing from /docs/static/ to /docs/public/
INFO: 2015/12/02 found taxonomies: map[string]string{"category":"categories", "tag":"tags"}
WARN: 2015/12/02 Unable to locate layout for section userguide: [section/userguide.html _default/section.html _default/list.html indexes/userguide.html _default/indexes.html theme/section/userguide.html theme/_default/section.html theme/_default/list.html theme/indexes/userguide.html theme/_default/indexes.html theme/section/userguide.html theme/_default/section.html theme/_default/list.html theme/indexes/userguide.html theme/_default/indexes.html theme/theme/section/userguide.html theme/theme/_default/section.html theme/theme/_default/list.html theme/theme/indexes/userguide.html theme/theme/_default/indexes.html]
WARN: 2015/12/02 Unable to locate layout for section misc: [section/misc.html _default/section.html _default/list.html indexes/misc.html _default/indexes.html theme/section/misc.html theme/_default/section.html theme/_default/list.html theme/indexes/misc.html theme/_default/indexes.html theme/section/misc.html theme/_default/section.html theme/_default/list.html theme/indexes/misc.html theme/_default/indexes.html theme/theme/section/misc.html theme/theme/_default/section.html theme/theme/_default/list.html theme/theme/indexes/misc.html theme/theme/_default/indexes.html]
WARN: 2015/12/02 Unable to locate layout for section reference: [section/reference.html _default/section.html _default/list.html indexes/reference.html _default/indexes.html theme/section/reference.html theme/_default/section.html theme/_default/list.html theme/indexes/reference.html theme/_default/indexes.html theme/section/reference.html theme/_default/section.html theme/_default/list.html theme/indexes/reference.html theme/_default/indexes.html theme/theme/section/reference.html theme/theme/_default/section.html theme/theme/_default/list.html theme/theme/indexes/reference.html theme/theme/_default/indexes.html]
WARN: 2015/12/02 Unable to locate layout for section examples: [section/examples.html _default/section.html _default/list.html indexes/examples.html _default/indexes.html theme/section/examples.html theme/_default/section.html theme/_default/list.html theme/indexes/examples.html theme/_default/indexes.html theme/section/examples.html theme/_default/section.html theme/_default/list.html theme/indexes/examples.html theme/_default/indexes.html theme/theme/section/examples.html theme/theme/_default/section.html theme/theme/_default/list.html theme/theme/indexes/examples.html theme/theme/_default/indexes.html]
WARN: 2015/12/02 Unable to locate layout for section articles: [section/articles.html _default/section.html _default/list.html indexes/articles.html _default/indexes.html theme/section/articles.html theme/_default/section.html theme/_default/list.html theme/indexes/articles.html theme/_default/indexes.html theme/section/articles.html theme/_default/section.html theme/_default/list.html theme/indexes/articles.html theme/_default/indexes.html theme/theme/section/articles.html theme/theme/_default/section.html theme/theme/_default/list.html theme/theme/indexes/articles.html theme/theme/_default/indexes.html]
WARN: 2015/12/02 Unable to locate layout for section security: [section/security.html _default/section.html _default/list.html indexes/security.html _default/indexes.html theme/section/security.html theme/_default/section.html theme/_default/list.html theme/indexes/security.html theme/_default/indexes.html theme/section/security.html theme/_default/section.html theme/_default/list.html theme/indexes/security.html theme/_default/indexes.html theme/theme/section/security.html theme/theme/_default/section.html theme/theme/_default/list.html theme/theme/indexes/security.html theme/theme/_default/indexes.html]
WARN: 2015/12/02 Unable to locate layout for section installation: [section/installation.html _default/section.html _default/list.html indexes/installation.html _default/indexes.html theme/section/installation.html theme/_default/section.html theme/_default/list.html theme/indexes/installation.html theme/_default/indexes.html theme/section/installation.html theme/_default/section.html theme/_default/list.html theme/indexes/installation.html theme/_default/indexes.html theme/theme/section/installation.html theme/theme/_default/section.html theme/theme/_default/list.html theme/theme/indexes/installation.html theme/theme/_default/indexes.html]
WARN: 2015/12/02 Unable to locate layout for section : [section/.html _default/section.html _default/list.html indexes/.html _default/indexes.html theme/section/.html theme/_default/section.html theme/_default/list.html theme/indexes/.html theme/_default/indexes.html theme/section/.html theme/_default/section.html theme/_default/list.html theme/indexes/.html theme/_default/indexes.html theme/theme/section/.html theme/theme/_default/section.html theme/theme/_default/list.html theme/theme/indexes/.html theme/theme/_default/indexes.html]
WARN: 2015/12/02 Unable to locate layout for section extend: [section/extend.html _default/section.html _default/list.html indexes/extend.html _default/indexes.html theme/section/extend.html theme/_default/section.html theme/_default/list.html theme/indexes/extend.html theme/_default/indexes.html theme/section/extend.html theme/_default/section.html theme/_default/list.html theme/indexes/extend.html theme/_default/indexes.html theme/theme/section/extend.html theme/theme/_default/section.html theme/theme/_default/list.html theme/theme/indexes/extend.html theme/theme/_default/indexes.html]
WARN: 2015/12/02 Unable to locate layout for section introduction: [section/introduction.html _default/section.html _default/list.html indexes/introduction.html _default/indexes.html theme/section/introduction.html theme/_default/section.html theme/_default/list.html theme/indexes/introduction.html theme/_default/indexes.html theme/section/introduction.html theme/_default/section.html theme/_default/list.html theme/indexes/introduction.html theme/_default/indexes.html theme/theme/section/introduction.html theme/theme/_default/section.html theme/theme/_default/list.html theme/theme/indexes/introduction.html theme/theme/_default/indexes.html]
0 of 17 drafts rendered
0 future content
169 pages created
117 non-page files copied
0 paginator pages created
0 tags created
0 categories created
in 562 ms
Serving pages from /docs/public
Web Server is available at http://localhost:8000/ (bind address 0.0.0.0)
Press Ctrl+C to stop
```

also see

```
find public/ -size 0
public/articles/index.html
public/reference/index.html
public/introduction/index.html
public/security/index.html
```

**Comment by anthonyfok** [comment]
See also Issue #947 - "Emit error/warning and suggested solution if a generated page is empty"

**Comment by SvenDowideit** [comment]
nice, thanks for noticing the duplication :)

**Comment by psmith1303** [comment]
I've had the problem with empty index.html files too ... this is where I am up to in my exploring.
The problem only happens for me with the FreeBSD version of Hugo
The problem doesn't happen with the Windows (64 bit) version.
One the Windows version using --verbose, I see:

`INFO: 2016/03/23 Alias "\blog\page\1" translated to "blog\page\1\index.html"`

This is what seems to be fixes the problem.

I also used to get the "WARN: 2015/12/02 Unable to locate layout for section" messages, but fixed that by creating some index.html files that had the following front matter in a folder called "content/bloglog".

```
---
type: bloglog
title: Archives
url: "/blog/archives"
---
```
Anyway, I hope that moves things forward a bit.

**Comment by mathwhiz1212** [comment]
I have the same issue, index.xml has all the right content but blank index.html files are being generated. I have two markdown files, index.md and a k.md neither of which were properly put it in a html file but they render properly and are placed into the index.xml file. Does anyone have a temporary fix?

**Comment by codefx9** [comment]
@mathwhiz1212, I wonder if https://github.com/spf13/hugo/issues/1072 might give you a work around.

**Comment by mathwhiz1212** [comment]
@codefx9 What am I supposed to put in empty_section.html?

**Comment by bep** [comment]
>@codefx9 What am I supposed to put in empty_section.html?

Nothing, as in ... empty.

**Comment by mathwhiz1212** [comment]
@bep Thanks. That mean that it didn't work because that is what I did.

**Comment by restlessdesign** [comment]
I ran into this issue (or a similar one) while trying to build my site using [Drone](drone.io) for my CI process. The root cause seemed to be that I either:

1. Edited my theme and left the theme submodule in a bad state
2. `git clone`d the theme and created a submodule in the first place

Deleting the submodule and copying the contents of the `themes/{THEME}/` back into a fresh theme folder did the trick for me!

**Comment by bep** [comment]
Should be fixed by 0406be3d547749b93dd4fd13e142982915b6ca1f
